### PR TITLE
Nxcm 4077 nexus download settings usertoken support

### PR DESCRIPTION
**Pull request by jdillon**
Adds support for pluggable interpolation, and includes $[userToken.] bits.

Pending support to encrypt w/master password muck.

### CODE REVIEW

**Comment by jdillon** [comment]
Master-password bits are now in, some details up here:

https://docs.sonatype.com/display/Nexus/User+Token+Testing#UserTokenTesting-nexus%3Asettingsdownload

File: nexus-maven-plugins/nexus-maven-plugin/src/main/java/org/sonatype/nexus/plugin/settings/usertoken/UserTokenDTO.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+//
+// This file was generated by the JavaTM Architecture for XML Binding(JAXB) Reference Implementation, v2.2.4 
+// See <a href="http://java.sun.com/xml/jaxb">http://java.sun.com/xml/jaxb</a> 
+// Any modifications to this file will be lost upon recompilation of the source schema. 
+// Generated on: 2012.05.18 at 02:12:38 PM PDT
```
Comment (nabcos) [inline nexus-maven-plugins/nexus-maven-plugin/src/main/java/org/sonatype/nexus/plugin/settings/usertoken/UserTokenDTO.java]: This does not look like it's still generated source. Needs a license header.

**Comment by nabcos** [comment]
+1

File: nexus-maven-plugins/nexus-maven-plugin/src/main/java/org/sonatype/nexus/plugin/settings/usertoken/UserTokenDTO.java (hunk 1 above)
Comment (jdillon) [inline nexus-maven-plugins/nexus-maven-plugin/src/main/java/org/sonatype/nexus/plugin/settings/usertoken/UserTokenDTO.java]: Thx, didn't notice (it started out as a copy, then I just removed all that muck)

**Comment by jdillon** [comment]
Minor changes to fix headers and ensure default settings from properties file are configured last when interpolating.

**Comment by peterlynch** [comment]
a few of these new classes could use some unit tests.

**Comment by jdillon** [comment]
BTW, I had been waiting on most work on tests until @brianf could review and validate this is indeed what the spec was intending, which I think he did yesterday.

**Comment by jdillon** [comment]
I've added a basic test for the customizer and interpolation bits.  Not sure there is much value in more UT for the other bits.  I'd add one for the encryption stuff except that the DefaultSecDispatcher in all its Plexus glory has private field injection of dependencies making it rather cumbersome to test with... and the code for the MasterPasswordEncryption stuff is/should be as close to what is in Maven, so I'd rather not refactor it much/or at all.

File: nexus-maven-plugins/nexus-maven-plugin/pom.xml [hunk 2]
```diff
@@ -358,7 +345,13 @@
...
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-interpolation</artifactId>
-      <version>1.13</version>
+      <version>1.15</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.sonatype.sisu.siesta</groupId>
+      <artifactId>siesta-client</artifactId>
+      <version>1.1-SNAPSHOT</version>
```
Comment (peterlynch) [inline nexus-maven-plugins/nexus-maven-plugin/pom.xml:354]: just in case siesta is now staged 1.1 and snap bumped to 1.2-SNAPSHOT

**Comment by peterlynch** [comment]
+1

**Pull request merged**